backend/vxlan/vxlan_network.go

```diff
@@ -111,140 +85,59 @@ type vxlanLeaseAttrs struct {
-	}
-
-	// "marked" events are skipped at the end.
-	eventMarker := make([]bool, len(batch))
-	leaseAttrsList := make([]vxlanLeaseAttrs, len(batch))
-	fdbEntryMarker := make([]bool, len(fdbTable))
-
-	// Run through the events "marking" ones that should be skipped
-	for eventMarkerIndex, evt := range batch {
-		if evt.Lease.Attrs.BackendType != "vxlan" {
-			log.Warningf("Ignoring non-vxlan subnet(%s): type=%v", evt.Lease.Subnet, evt.Lease.Attrs.BackendType)
-			eventMarker[eventMarkerIndex] = true
+		if event.Lease.Attrs.BackendType != "vxlan" {
+			log.Warningf("Ignoring non-vxlan subnet(%s): type=%v", event.Lease.Subnet, event.Lease.Attrs.BackendType)
 			continue
 		}
 
-		// Parse the vxlan specific backend data
-		if err := json.Unmarshal(evt.Lease.Attrs.BackendData, &leaseAttrsList[eventMarkerIndex]); err != nil {
+		var attrs vxlanLeaseAttrs
+		if err := json.Unmarshal(event.Lease.Attrs.BackendData, &attrs); err != nil {
 			log.Error("Error decoding subnet lease JSON: ", err)
-			eventMarker[eventMarkerIndex] = true
 			continue
 		}
 
-		// Check the existing VTEP->Public IP mappings.
-		// If there's already an entry with the right VTEP and Public IP then the event can be skipped and the FDB entry can be retained
-		for j, fdbEntry := range fdbTable {
-			if evt.Lease.Attrs.PublicIP.ToIP().Equal(fdbEntry.IP) && bytes.Equal([]byte(leaseAttrsList[eventMarkerIndex].VtepMAC), []byte(fdbEntry.HardwareAddr)) {
-				eventMarker[eventMarkerIndex] = true
-				fdbEntryMarker[j] = true
-				break
-			}
+		route := netlink.Route{
+			LinkIndex: nw.dev.link.Attrs().Index,
+			Scope:     netlink.SCOPE_UNIVERSE,
+			Dst:       event.Lease.Subnet.ToIPNet(),
+			Gw:        event.Lease.Subnet.IP.ToIP(),
 		}
+		route.SetFlag(syscall.RTNH_F_ONLINK)
 
-		// Store off the subnet lease and VTEP
-		nw.routes.set(evt.Lease.Subnet, net.HardwareAddr(leaseAttrsList[eventMarkerIndex].VtepMAC))
-		log.V(2).Infof("Adding subnet: %s PublicIP: %s VtepMAC: %s", evt.Lease.Subnet, evt.Lease.Attrs.PublicIP, net.HardwareAddr(leaseAttrsList[eventMarkerIndex].VtepMAC))
-	}
+		switch event.Type {
+		case subnet.EventAdded:
+			log.V(2).Infof("Adding subnet: %s PublicIP: %s VtepMAC: %s", event.Lease.Subnet, event.Lease.Attrs.PublicIP, net.HardwareAddr(attrs.VtepMAC))
+			if err := nw.dev.AddArp(neighbor{IP: event.Lease.Subnet.IP, MAC: net.HardwareAddr(attrs.VtepMAC)}); err != nil {
+				log.Error("AddArp failed: ", err)
+			}
 
-	// Loop over the existing FDB entries, deleting any that shouldn't be there
-	for j, marker := range fdbEntryMarker {
-		if !marker && fdbTable[j].IP != nil {
-			err := nw.dev.DelL2(neighbor{IP: ip.FromIP(fdbTable[j].IP), MAC: fdbTable[j].HardwareAddr})
-			if err != nil {
-				log.Error("Delete L2 failed: ", err)
+			if err := nw.dev.AddFdb(neighbor{IP: event.Lease.Attrs.PublicIP, MAC: net.HardwareAddr(attrs.VtepMAC)}); err != nil {
+				log.Error("AddFdb failed: ", err)
```

Comment:
gotcha, I thing maybe you're suggesting a "continue" rather than "return"? And advocating for cleaning up the failed FDB or arp entries? It sounds like a reasonable suggestion to me (though I don't think it really brings much benefit to the user and increases the code complexity).